### PR TITLE
fix(API): pass authMode used for lazy loading functionality

### PR DESCRIPTION
**Pull request by lawmicha**
## Issue \#

https://github.com/aws-amplify/amplify-swift/issues/3677

## Description

Lazy Loading functionality is built on metadata passed to it during decoding. For example, the hasMany list will contain the parent's identifiers and the apiName to reconstruct the request when lazy loading. The belongsTo lazy model will contain its identifiers and the apiName used to when the child model was retrieved. 

The issue is with the new `authMode` parameter, the `authMode` was not being passed down to the lazy types (https://github.com/aws-amplify/amplify-swift/issues/3677#issuecomment-2108404457)

```swift
// default auth type is APIKey.
let message = try await Amplify.API.query(request: .get(Message.self, byId: id, authMode: .amazonCognitoUserPools)).get() // use userPool auth
let sender = try await message.sender  // with fix, uses user pool auth. without fix, uses default API key auth from config
let space = try await message.space // with fix, uses user pool auth. without fix, uses default API key auth from config
```

## General Checklist

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/API/Sources/AWSAPIPlugin/Core/AppSyncListPayload.swift [hunk 1]
```diff
@@ -43,7 +46,7 @@ public struct AppSyncListPayload: Codable {
         // `GraphQLFilter`.
         guard let filterVariablesData = try? encoder.encode(filterValue),
               let filterVariablesJSON = try? JSONSerialization.jsonObject(with: filterVariablesData)
-                as? GraphQLFilter else {
+                as? [String: Any] else {
```
Comment (thisisabhash) [inline AmplifyPlugins/API/Sources/AWSAPIPlugin/Core/AppSyncListPayload.swift:49]: why this change?

**Review by thisisabhash — DISMISSED** [review]

File: AmplifyPlugins/API/Sources/AWSAPIPlugin/Core/AppSyncListPayload.swift (hunk 1 above)
Comment (lawmicha) [reply AmplifyPlugins/API/Sources/AWSAPIPlugin/Core/AppSyncListPayload.swift]: i've reverted it, when i was compiling earlier Xcode wasn't picking up GraphQLFilter properly from AWSPluginsCore

**Review by thisisabhash — APPROVED** [review]
